csrc/python_frontend/python_bindings.cpp

```diff
@@ -149,6 +149,46 @@ Tensor reshape_fn(
   return output;
 }
 
+template <class ShapeType, serde::RecordType RType>
+Tensor random_dist_op_fn(
+    FusionDefinition::Operators& self,
+    Scalar arg1,
+    Scalar arg2,
+    ShapeType generic_new_shape,
+    std::optional<Scalar> rng_seed,
+    std::optional<Scalar> rng_offset,
+    PrimDataType dtype) {
+  static_assert(
+      (RType == serde::RecordType_NormalDistOp) ||
+      (RType == serde::RecordType_UniformDistOp));
+  NVF_CHECK(self.validUse(), "Attempting to add to a completed definition!");
+  NVF_CHECK(
+      isFloatingPointType(dtype),
+      "Random distributions only create floating point types! ",
+      dtype);
+  FusionDefinition* fd = self.fusion_definition;
+  Vector new_shape = ShapeAsVector(generic_new_shape, *fd);
+
+  Tensor output = fd->defineTensor(new_shape.size);
+  std::vector<State> arg_states = {
+      fd->recordingState(arg1()),
+      fd->recordingState(arg2()),
+      fd->recordingState(new_shape()),
+  };
+  if (rng_seed.has_value()) {
```

Comment:
nitpick. should we raise error when rng_offset is provided but rng_seed is missing?

Reply:
Good idea!